src/js/jsx/shared/Datalist.jsx

```diff
@@ -340,9 +357,15 @@ define(function (require, exports, module) {
                     break;
                 case "Enter":
                 case "Return":
-                    select.close(event, "apply");
-                    if (dialog && dialog.isOpen()) {
-                        dialog.toggle(event);
+                    var noOptionsExecute = optionsList ? optionsList.toJS()[0].handleExecute : null;
```

Comment:
I don't think that `Datallist` should be doing anything that involves `handleExecute`, which is only relevant to `Search` AFAIK. Can you help me understand what this is used for so we can find a way to move this functionality out of `Datalist`?